components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.email/src/main/java/org/wso2/carbon/event/output/adapter/email/EmailEventAdapter.java

```diff
@@ -232,22 +243,24 @@ public void connect() throws ConnectionUnavailableException {
             try {
                 smtpFromAddress = new InternetAddress(smtpFrom);
             } catch (AddressException e) {
-                String msg = "failed to connect to the mail server due to error in retrieving " +
-                        "smtp from address";
+                String msg = "failed to connect to the mail server due to error in retrieving smtp from address";
                 if (log.isDebugEnabled()) {
                     log.debug(msg + ": " + smtpFrom);
                 }
                 throw new ConnectionUnavailableException
                         ("The adapter " + eventAdapterConfiguration.getName() + " " + msg, e);
             }
 
-            //Retrieving username and password of SMTP server.
-            smtpUsername = props.getProperty(MailConstants.MAIL_SMTP_USERNAME);
-            smtpPassword = props.getProperty(MailConstants.MAIL_SMTP_PASSWORD);
-
+            Map<String, String> credentials = resolveCredentials(globalProperties);
+            smtpUsername = credentials.get(USERNAME);
+            smtpPassword = credentials.get(PASSWORD);
```

Comment:
We need to use a char array variable to store passwords as the strings are immutable, it will be available until the GC happens and it can detect the password when analyzing a heap dump.